File: server/src/models/review.ts
import { ObjectId } from 'mongodb';

export interface Review {
    _id: ObjectId;
    text: string;
    name: string;
    rating?: number; // Optional as it could be added in a NLP Lab
    timestamp: number;

    /**
     * Reference to the book collection.
     */
    bookId: string;
}

File: java-server/src/main/java/com/mongodb/devrel/library/domain/model/Review.java
package com.mongodb.devrel.library.domain.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "reviews")
public record Review(
        @Id
        @JsonProperty("_id")
        String id,
        String text,
        Integer rating,
        String name,
        Object timestamp,
        String bookId
){}